prov/gni/src/gnix_nic.c

```diff
@@ -81,14 +300,14 @@ static int __gnix_nic_tx_freelist_init(struct gnix_nic *nic, int n_descs)
 		goto err;
 	}
 
-	list_head_init(&nic->tx_desc_free_list);
-	list_head_init(&nic->tx_desc_active_list);
+	dlist_init(&nic->tx_desc_free_list);
+	dlist_init(&nic->tx_desc_active_list);
 
-	for (i = 0, desc_ptr = desc_base; i < n_descs; i++) {
+	for (i = 0, desc_ptr = desc_base; i < n_descs; i++, desc_ptr++) {
 		desc_ptr->desc.id = i;
-		gnix_list_node_init(&desc_ptr->desc.list);
-		list_add_tail(&nic->tx_desc_free_list,
-				&desc_ptr->desc.list);
+		dlist_init(&desc_ptr->desc.list);
```

Comment:
I actually don't think this matters, but the corresponding dlist call is `dlist_node_init`.  @hppritcha Is there a reason for making this call?  Seems like ithe pointers are just going to be reset by the call to `dlist_insert_tail` on the next line.  I also left it as is in `_gnix_dgram_hndl_alloc`, but maybe that call should go away also.